DIPs/DIP1002.md

```diff
@@ -0,0 +1,90 @@
+# TryElseExpression
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | 1002                                                            |
+| Author:         | [Sophie Kirschner](https://github.com/pineapplemachine)         |
+| Implementation: | n/a                                                             |
+| Status:         | Draft                                                           |
+
+## Abstract
+
+In Python, the try/catch/finally syntax is augmented with an additional clause,
+termed else. It is a fantastically useful addition to the conventional syntax.
+It works like this:
+
+``` python
+    try:
+        do_something()
+    except Exception as e:
+        pass # Runs when an error inheriting from Exception was raised
+    else:
+        pass # Runs when no error was raised
+    finally:
+        pass # Runs unconditionally, evaluates last
+```
+
+### Links
+
+* [Exception handling in Python](https://docs.python.org/3/tutorial/errors.html#handling-exceptions)
+
+## Description
+
+The _TryStatement_ would gain an additional piece, _TryElseStatement_, which
+should be present after all _Catches_ and before a _FinallyStatement_, if any.
+
+The code in a _TryElseStatement_ would occur after the try's _ScopeStatement_,
+would occur only if no exception was thrown, and would occur before the
+_FinallyStatement_, if any is present. If any error occurs inside a
+_TryElseStatement_, its corresponding _FinallyStatement_ is still evaluated.
+
+### Rationale
+
+Having such a facility built in to D's exception handling will allow the writing
+of cleaner, less user-error-prone code.
+
+### Breaking changes / deprecation process
+
+This addition should not break any existing code.
+
+### Examples
+
+``` D
+try{
+    do_a_thing();
+}catch(Exception exception){
+    handle_error();
+}else{
+    depends_on_success_of_thing();
+}finally{
+    do_this_always();
+}
+```
+
+This is equivalent to:
+
+``` D
+bool success = false;
+try{
+    do_a_thing();
+    success = true;
+}catch(Exception exception){
+    handle_error();
+}finally{
+    try{
+        if(success){
+            depends_on_success_of_thing();
+        }
+    }finally{
+        do_this_always();
+    }
+}
+```
```

Comment:
As far as convincing arguments go, I'm not sure how much better it gets than "this is a proven and commonly-used pattern present in a prominent language, and it can be added to D without breaking any existing code."